src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,106 @@
+package duke;
+
+import duke.task.Task;
+
+import java.io.Serializable;
+import java.sql.Array;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Represents the list of user's tasks.
+ */
+public class TaskList implements Serializable {
+
+    private ArrayList<Task> tasks;
+
+    /**
+     * Creates a new empty list of tasks.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Initialize based on the list of tasks saved in the user's memory.
+     *
+     * @param tasks The saved list of tasks.
+     */
+    public TaskList(TaskList tasks) {
+        this.tasks = new ArrayList<>(tasks.tasks);
+    }
+
+    /**
+     * Marks the task corresponding to the index as completed.
+     *
+     * @param index The index of the task to be marked completed.
+     * @return The completed task.
+     */
+    public Task mark(int index) {
+        Task task = this.tasks.get(index);
```

Comment:
I am unsure if "this" statement is necessary?

This doesn't recommend based on the coding standard.

Can considering removing this statement.

Reply:
I wasn't aware of this, my perception was that the "this" keyword should be used where possible to make the code more explicit. I tried finding information related to the "this" keyword in the coding standard document @ https://se-education.org/guides/conventions/java/intermediate.html but couldn't find it, mind pointing me in the right direction? thanks 😄